pandas/tests/series/indexing/test_datetime.py

```diff
@@ -667,3 +667,16 @@ def test_setitem_tuple_with_datetimetz():
     result[(0, 1)] = np.nan
     expected.iloc[0] = np.nan
     tm.assert_series_equal(result, expected)
+
+
+"""
+BUG: indexing regression with datetime index
+"""
+
+
+def test_indexing_regression():
+    arr = date_range("1/1/2008", "1/2/2008")
+    ser = arr.to_series()["2008"]
+    assert type(ser) == Series
```

Comment:
Instead of these asserts use `tm.assert_series_equal` and construct the expected series